src/modules/auth/pipe/password-validation.pipe.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable, PipeTransform, BadRequestException } from '@nestjs/common';
+import * as Joi from 'joi';
+import { passwordPatten } from '@/common/utils/helpers';
+
+@Injectable()
+export class PasswordValidationPipe implements PipeTransform {
+  private readonly schema = Joi.string().pattern(passwordPatten).messages({
+    'string.pattern.base': 'password must contain 8 to 32 characters, including letter, number and special character.'
+  });
+
+  transform(value: any): string {
```

Comment:
consider to use custom exception to give non-error response, 
now we get:
<img width="716" alt="image" src="https://github.com/BeeQuantAI/platform_api/assets/1610136/910d7532-e403-499d-8dec-265e1726b812">
here's like what we want:
<img width="509" alt="image" src="https://github.com/BeeQuantAI/platform_api/assets/1610136/4ac1724d-e676-4335-843f-15a8494ace1f">

Reply:
![Screenshot 2024-05-29 at 17 46 10](https://github.com/BeeQuantAI/platform_api/assets/48723564/fea9d00a-4595-469f-b575-dce4f7d8a9a7)